### PR TITLE
[DEE-440] added error msg to checks that do not support token/multi-label classification

### PR DESCRIPTION
**Pull request by yromanyshyn**

### CODE REVIEW

File: deepchecks/nlp/checks/data_integrity/property_label_correlation.py [hunk 1]
```diff
@@ -92,8 +92,18 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         DeepchecksValueError
             If the object is not a Dataset instance with a label.
         """
-        text_data = context.get_data_by_kind(dataset_kind).sample(self.n_samples, random_state=context.random_state)
+        type_name = type(self).__name__
 
+        if context.task_type is TaskType.TOKEN_CLASSIFICATION:
+            task_type_name = TaskType.TOKEN_CLASSIFICATION.value
+            raise DeepchecksNotSupportedError(f'"{type_name}" is not suited for the "{task_type_name}" tasks')
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/data_integrity/property_label_correlation.py:99]: Could be more elegant to use a common function like we have the assert_classification_task() in tabular

Comment (Nadav-Barak) [reply deepchecks/nlp/checks/data_integrity/property_label_correlation.py:99]:
```suggestion
            raise DeepchecksNotSupportedError(f'"{type_name}" is not supported for the "{task_type_name}" tasks')
```

File: deepchecks/nlp/checks/data_integrity/property_label_correlation.py [hunk 2]
```diff
@@ -92,8 +92,18 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
...
+
+        text_data = context.get_data_by_kind(dataset_kind)
+
+        if text_data.is_multi_label_classification():
```
Comment (Nadav-Barak) [inline deepchecks/nlp/checks/data_integrity/property_label_correlation.py:103]: same as above - maybe can even be under the same function?

File: deepchecks/nlp/context.py [hunk 3]
```diff
@@ -216,18 +216,47 @@ def _validate_classification_prediction(dataset: TextData, prediction: TTextPred
...
-                raise ValidationError(f'Check requires predictions for {dataset.name} to be a sequence '
-                                      f'of sequences of strings or integers')
-            if len(prediction[i]) != len(dataset.tokenized_text[i]):
-                raise ValidationError(f'Check requires predictions for {dataset.name} to have '
-                                      f'the same number of tokens as the input text')
+        if (
+            isinstance(prediction, (str, bytes, bytearray))
+            or not isinstance(prediction, (np.ndarray, t.Sequence))
+            # TODO: fix error messages, make it more particular
+            # third party containers do not extend Sequence. Example:
```
Comment (Nadav-Barak) [inline deepchecks/nlp/context.py:223]: agree - see is_sequence_not_str function as alternative

Comment (Nadav-Barak) [reply deepchecks/nlp/context.py:223]: In general the NLP dummy model is not that good IMO (in the future it should be updated or removed - see https://linear.app/deepchecks/issue/DEE-456/unify-dummy-models)

File: deepchecks/nlp/metric_utils/token_classification.py [hunk 4]
```diff
@@ -21,15 +21,19 @@
 __all__ = ['get_default_token_scorers', 'validate_scorers', 'get_scorer_dict']
 
 DEFAULT_AVG_SCORER_NAMES = ('f1_macro', 'recall_macro', 'precision_macro')
-DEFAULT_PER_CLASS_SCORER_NAMES = ('f1_per_class', 'f1_per_class', 'f1_per_class')
+DEFAULT_PER_CLASS_SCORER_NAMES = tuple()
```
Comment (Nadav-Barak) [inline deepchecks/nlp/metric_utils/token_classification.py:24]: why empty?

File: deepchecks/nlp/checks/data_integrity/property_label_correlation.py [hunk 5]
```diff
@@ -92,8 +92,18 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
...
+        type_name = type(self).__name__
 
+        if context.task_type is TaskType.TOKEN_CLASSIFICATION:
+            task_type_name = TaskType.TOKEN_CLASSIFICATION.value
+            raise DeepchecksNotSupportedError(f'"{type_name}" is not supported for the "{task_type_name}" tasks')
+
+        text_data = context.get_data_by_kind(dataset_kind)
+
+        if text_data.is_multi_label_classification():
+            raise DeepchecksNotSupportedError(f'"{type_name}" is not suited for the multilable classification tasks')
```
Comment (nirhutnik) [inline deepchecks/nlp/checks/data_integrity/property_label_correlation.py:104]:
```suggestion
            raise DeepchecksNotSupportedError(f'"{type_name}" is not supported for the multilable classification tasks')
```

Comment (nirhutnik) [reply deepchecks/nlp/checks/data_integrity/property_label_correlation.py:104]: And in the rest of the places where it says "suited" :)

File: deepchecks/nlp/context.py (hunk 3 above)
Comment (yromanyshyn) [reply deepchecks/nlp/context.py]: replaced with `is_sequence_not_str`

File: deepchecks/nlp/checks/data_integrity/property_label_correlation.py (hunk 1 above)
Comment (yromanyshyn) [reply deepchecks/nlp/checks/data_integrity/property_label_correlation.py]: fixed

File: deepchecks/nlp/checks/data_integrity/property_label_correlation.py (hunk 2 above)
Comment (yromanyshyn) [reply deepchecks/nlp/checks/data_integrity/property_label_correlation.py]: fixed,
I leave them seperate

File: deepchecks/nlp/context.py [hunk 6]
```diff
@@ -430,6 +446,30 @@ def assert_properties(text_data):
...
+        if self.task_type is TaskType.TOKEN_CLASSIFICATION:
+            raise DeepchecksNotSupportedError(
+                f'"{check_name}" is not supported for the "{task_type_name}" tasks'
+            )
+
+    def raise_if_multi_label_task(self, check=None):
+        """Raise an exception if it is a multi-label classification task."""
+        dataset = self._train if self._train is not None else self._test
+
+        if dataset is None:
```
Comment (Nadav-Barak) [inline deepchecks/nlp/context.py:462]: redundant - validated on creation of context

File: deepchecks/nlp/context.py [hunk 7]
```diff
@@ -430,6 +446,30 @@ def assert_properties(text_data):
...
+            return
+
+        check_name = type(check).__name__ if check else 'Check'
+        is_multi_label = dataset.is_multi_label_classification()
+
+        if is_multi_label:
```
Comment (Nadav-Barak) [inline deepchecks/nlp/context.py:468]:
```suggestion
        if dataset.is_multi_label_classification():
```

**Review by Nadav-Barak — APPROVED** [review]

File: deepchecks/nlp/context.py (hunk 6 above)
Comment (yromanyshyn) [reply deepchecks/nlp/context.py]: removed

File: deepchecks/nlp/checks/data_integrity/property_label_correlation.py [hunk 8]
```diff
@@ -92,8 +92,11 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         DeepchecksValueError
             If the object is not a Dataset instance with a label.
         """
-        text_data = context.get_data_by_kind(dataset_kind).sample(self.n_samples, random_state=context.random_state)
+        context.raise_if_token_classification_task(self)
+        context.raise_if_multi_label_task(self)
 
+        text_data = context.get_data_by_kind(dataset_kind)
+        text_data = text_data.sample(self.n_samples, random_state=context.random_state)
```
Comment (JKL98ISR) [inline deepchecks/nlp/checks/data_integrity/property_label_correlation.py:99]: why split to 2 lines?

**Review by JKL98ISR — APPROVED** [review]

**Pull request merged**